components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java

```diff
@@ -23,44 +23,230 @@
+                String propertyValue = businessOwnerPropertiesDTO.getValue();
+                Boolean isVisible = businessOwnerPropertiesDTO.getIsVisible();
+                if (StringUtils.isEmpty(propertyId) || StringUtils.isEmpty(propertyValue) || isVisible == null) {
+                    RestApiUtil.handleBadRequest("Business owner properties cannot be empty or null.", log);
+                }
+                businessOwnerProperty.setPropertyId(propertyId.trim());
+                businessOwnerProperty.setPropertyValue(propertyValue.trim());
+                businessOwnerProperty.setShowingInStore(isVisible);
+                businessOwnerPropertyList.add(businessOwnerProperty);
+            }
+            businessOwner.setBusinessOwnerPropertiesList(businessOwnerPropertyList);
+
+            //save business owner.
+            int ownerId = apiProvider.saveBusinessOwner(businessOwner);
+
+            //Retrieve the added business owner to send in the response payload.
+            BusinessOwner addedBusinessOwner = apiProvider.getBusinessOwner(ownerId);
+            businessOwnerDTO.setId(ownerId);
+            businessOwnerDTO.setName(addedBusinessOwner.getBusinessOwnerName());
+            businessOwnerDTO.setEmail(addedBusinessOwner.getBusinessOwnerEmail());
+            businessOwnerDTO.setDescription(addedBusinessOwner.getBusinessOwnerDescription());
+            businessOwnerDTO.setSite(addedBusinessOwner.getBusinessOwnerSite());
+
+            List<BusinessOwnerProperty> ownerPropertyList = businessOwner.getBusinessOwnerPropertiesList();
+            List<BusinessOwnerPropertiesDTO> ownerPropertiesDTOList = new ArrayList<>();
+            //save custom properties of the owner.
+            for (BusinessOwnerProperty ownerProperty : ownerPropertyList) {
+                BusinessOwnerPropertiesDTO businessOwnerPropertiesDTO = new BusinessOwnerPropertiesDTO();
+                businessOwnerPropertiesDTO.setKey(ownerProperty.getPropertyId());
+                businessOwnerPropertiesDTO.setValue(ownerProperty.getPropertyValue());
+                businessOwnerPropertiesDTO.setIsVisible(ownerProperty.isShowingInStore());
+                ownerPropertiesDTOList.add(businessOwnerPropertiesDTO);
+            }
+            businessOwnerDTO.setProperties(ownerPropertiesDTOList);
+        } catch (AppManagementException e) {
+            String errorMessage = "Error while saving Business Owner.";
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return Response.ok().entity(businessOwnerDTO).build();
     }
 
     @Override
     public Response administrationBusinessownerBusinessOwnerIdGet(String businessOwnerId, String accept, String ifNoneMatch) {
-        return null;
+        BusinessOwnerDTO businessOwnerDTO = new BusinessOwnerDTO();
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            //get policy details related to id
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0